api/base/middleware.py

```diff
@@ -47,7 +49,10 @@ def process_response(self, request, response):
         except OperationFailure as err:
             message = utils.get_error_message(err)
             if messages.NO_TRANSACTION_TO_COMMIT_ERROR not in message:
-                raise err
+                if settings.DEBUG_TRANSACTIONS:
```

Comment:
Can you please clarify:
a) What the purpose of this code is, and
b) Whether it is related to this PR? (I'm trying to make sure local-only debug changes didn't accidentally make it into this commit)

In particular, this seems like a significant enough change to transactions that it should at LEAST be described separately. I'm not sold on the utility of a DEBUG setting that suppresses warnings altogether; if the issue isn't at least logged, it seems like this would make debugging harder, instead of easier.